src/backend/distributed/utils/maintenanced.c

```diff
@@ -326,6 +326,7 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 			{
 				WarnIfSyncDNS();
 				statsCollectionSuccess = CollectBasicUsageStatistics();
+				CheckForUpdates();
```

Comment:
In the—admittedly weird—case where usage statistics fails but an update check succeeds, we would check for updates twice in a minute. Is that OK?

Maybe the `CheckForUpdates` call could be moved into the `statsCollectionSuccess` block (but with a suitable comment to explain why)?

Reply:
Done.